warrior/WarriorCore/testsuite_driver.py

```diff
@@ -320,12 +264,59 @@ def execute_testsuite(testsuite_filepath, data_repository, from_project,
-
-    elif execution_type.upper() == 'RUN_UNTIL_FAIL':
+        if runmode is None:
+            print_info("Executing testccases sequentially")
+            test_suite_status = sequential_testcase_driver.main(testcase_list, suite_repository,
+                                                                data_repository, from_project,
+                                                                auto_defects=auto_defects)
+        elif runmode.upper() == 'RMT':
+
+            print_info("Execution type: {0}, Max Attempts: {1}".format(execution_type, value))
+
+            i = 0
+            while i < int(value):
+                i += 1
+                print_debug("\n\n<======= ATTEMPT: {0} ======>".format(i))
+                # We aren't actually summing each test result here...
+                test_suite_status = sequential_testcase_driver.main(testcase_list, suite_repository,
+                                                                    data_repository,
+                                                                    from_project,
+                                                                    auto_defects=auto_defects)
+
+
+        elif runmode.upper() == 'RUP':
+            print_info("Execution type: {0}, Attempts: {1}".format(execution_type, value))
+            i = 0
+            while i < int(value):
+                i += 1
+                print_debug("\n\n<======= ATTEMPT: {0} ======>".format(i))
+                test_suite_status = sequential_testcase_driver.main(testcase_list, suite_repository,
+                                                                    data_repository, from_project,
+                                                                    auto_defects=auto_defects)
+                test_count = i * len(testcase_list)
+                testsuite_utils.pSuite_update_suite_tests(str(test_count))
+                if str(test_suite_status).upper() == "TRUE":
+                    break
+
+        elif runmode.upper() == 'RUF':
+
+            print_info("Execution type: {0}, Attempts: {1}".format(execution_type, value))
+            i = 0
+            while i < int(value):
+                i += 1
+                print_debug("\n\n<======= ATTEMPT: {0} ======>".format(i))
+                test_suite_status = sequential_testcase_driver.main(testcase_list, suite_repository,
+                                                                    data_repository, from_project,
+                                                                    auto_defects=auto_defects)
+                test_count = i * len(testcase_list)
+                testsuite_utils.pSuite_update_suite_tests(str(test_count))
+                if str(test_suite_status).upper() == "FALSE" or \
+                                str(test_suite_status).upper() == "ERROR":
+                    break
+
+    #The below ELIF is to preserve backward compatibility. The new logic is available/
```

Comment:
If the value of execution_type hasn't changed, why would the runmode logic not handled by the original code?

Reply:
The Runmode is under execution type in the current design. So, I am not changing anything to the current logic to preserve backward compatibility. To bring uniformity across runmodes, I have introduced a tag similar to the tags available for runmode in suite_step, Project_step & Testcase_step in the Suite Global tags. Going forward users would see the new tags through Katana (This would be similar to all runmode options across Warrior). 
If this explanation doesn't help, I shall call you tomorrow to discuss.